backend/kernelCI_app/views/bootsView.py

```diff
@@ -0,0 +1,126 @@
+        if error_message:
+            return error_message
+        print("unknown error_msg in misc", misc)
+        return "unknown error"
+
+    def get(self, request, commit_hash: str | None):
+        names_map = {
+            "c.id": "id",
+            "c.git_repository_url": "git_repository_url",
+            "c.git_commit_hash": "git_commit_hash",
+            "t.build_id": "build_id",
+            "t.start_time": "start_time",
+            "t.status": "status",
+            "t.path": "path",
+            "b.architecture": "architecture",
+            "b.config_name": "config_name",
+            "b.compiler": "compiler",
+            "t.environment_misc": "environment_misc",
+            "t.environment_comment": "environment_comment",
+            "t.misc": "misc",
+        }
+
+        query = Checkouts.objects.raw(
+            """
+                SELECT DISTINCT ON (t.build_id) c.id, c.git_repository_url,
+                c.git_commit_hash, t.build_id, t.start_time,
+                t.status as status, t.path, b.architecture, b.config_name,
+                b.compiler, t.environment_misc, t.environment_comment, t.misc FROM checkouts AS c
+                INNER JOIN builds AS b ON c.id = b.checkout_id
+                INNER JOIN tests AS t ON t.build_id = b.id
+                WHERE c.git_commit_hash = %s AND t.path LIKE 'boot%%'
+                ORDER BY
+                build_id,
+                CASE t.status
+                    WHEN 'FAIL' THEN 1
+                    WHEN 'ERROR' THEN 2
+                    WHEN 'MISS' THEN 3
+                    WHEN 'PASS' THEN 4
+                ELSE 4
+                END,
+                t.id;
+            """,
+            [commit_hash],
+            translations=names_map,
+        )
+
+        errorCounts = {}
+        configCounts = {}
+        bootHistory = []
+        errorCountPerArchitecture = {}
+        platforms = set()
+        compilersPerArchitecture = {}
+        errorMessageCounts = {}
+        for record in query:
+            errorCounts = defaultdict(int)
+            errorCounts[record.status] += 1
+            configCounts = defaultdict(int)
+            configCounts[record.config_name] += 1
+            bootHistory.append(
+                {"start_time": record.start_time, "status": record.status}
+            )
+            if (
+                record.status == "MISS"
+                or record.status == "ERROR"
+                or record.status == "FAIL"
+            ):
+                errorCountPerArchitecture = defaultdict(int)
+                errorCountPerArchitecture[record.architecture] += 1
+                compilersPerArchitecture = defaultdict(set)
+                compilersPerArchitecture[record.architecture].add(record.compiler)
+                platforms.add(self.extract_platform(record.environment_misc))
+                currentErrorMessage = self.extract_error_message(record.misc)
+                errorMessageCounts = defaultdict(int)
```

Comment:
you misunderstood the use of defaultdict

```suggestion
        errorCounts = defaultdict(int)
        configCounts =  defaultdict(int)
        bootHistory = []
        errorCountPerArchitecture =  defaultdict(int)
        platforms = set()
        compilersPerArchitecture = defaultdict(set)
        errorMessageCounts = defaultdict(int)
        for record in query:
            errorCounts[record.status] += 1
            configCounts[record.config_name] += 1
            bootHistory.append(
                {"start_time": record.start_time, "status": record.status}
            )
            if (
                record.status == "MISS"
                or record.status == "ERROR"
                or record.status == "FAIL"
            ):
                errorCountPerArchitecture[record.architecture] += 1
                compilersPerArchitecture[record.architecture].add(record.compiler)
                platforms.add(self.extract_platform(record.environment_misc))
                currentErrorMessage = self.extract_error_message(record.misc)
                errorMessageCounts[currentErrorMessage] += 1
```

Reply:
Oh, I didn't misunderstood, I just didn't think of that, thx